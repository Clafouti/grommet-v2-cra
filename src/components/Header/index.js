import React from 'react';
import PropTypes from 'prop-types';
import Box from 'grommet/components/Box';

export const Header = ({ children }) =>
  <Box
    tag='header'
    background='brand'
    align='center'
    pad={{
      horizontal: 'medium'
    }}
  >
    {children}
  </Box>;

Header.propTypes = {
  children: PropTypes.node
};

export default Header;
