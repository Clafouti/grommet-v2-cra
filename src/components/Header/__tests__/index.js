import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Header from '../../Header';

describe('<Header />', () => {
  it('renders a Header', () => {
    const wrapper = shallow(
      <Header />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('renders a Header with children', () => {
    const wrapper = shallow(
      <Header>
        <p>test!</p>
      </Header>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
