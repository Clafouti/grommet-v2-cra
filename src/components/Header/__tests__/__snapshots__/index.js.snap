// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders a Header 1`] = `
<Box
  align="center"
  background="brand"
  pad={
    Object {
      "horizontal": "medium",
    }
  }
  tag="header"
/>
`;

exports[`<Header /> renders a Header with children 1`] = `
<Box
  align="center"
  background="brand"
  pad={
    Object {
      "horizontal": "medium",
    }
  }
  tag="header"
>
  <p>
    test!
  </p>
</Box>
`;
